tests/tt_metal/tt_metal/perf_microbenchmark/ethernet/test_workers_and_erisc_datamover_unidirectional.cpp

```diff
@@ -0,0 +1,622 @@
+
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <algorithm>
+#include <functional>
+#include <limits>
+#include <random>
+
+#include "tt_backend_api_types.hpp"
+#include "tt_metal/common/core_coord.h"
+#include "tt_metal/common/math.hpp"
+#include "tt_metal/detail/tt_metal.hpp"
+#include "tt_metal/host_api.hpp"
+#include "tt_metal/impl/kernels/kernel.hpp"
+#include "tt_metal/test_utils/comparison.hpp"
+#include "tt_metal/test_utils/df/df.hpp"
+#include "tt_metal/test_utils/env_vars.hpp"
+#include "tt_metal/test_utils/print_helpers.hpp"
+#include "tt_metal/test_utils/stimulus.hpp"
+
+// #include "impl/kernels/kernel_types.hpp"
+
+using namespace tt;
+using namespace tt::test_utils;
+using namespace tt::test_utils::df;
+
+class N300TestDevice {
+   public:
+    N300TestDevice() : device_open(false) {
+        auto slow_dispatch = getenv("TT_METAL_SLOW_DISPATCH_MODE");
```

Comment:
Are you able to try out your tests with FD on t3000 pci chips now? Ultimate benchmark is very good bandwidth with tunneling + all the apis here 🙏

Reply:
Will do!